### PR TITLE
Add method to get InputStream from channel

### PR DESCRIPTION
**Pull request by wggihan**
## Purpose
Get a InputStram from the channel for a content passthrough scenarios.

## Goals
Either Blob or String array will return current channel implementation. But in a content passthrough scenarios like upload a large attachment to a remote FTP server, we should able to transfer content without building in the server.

## Approach
Get an input stream from the source and pass that pointer to necessary client.

## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

### CODE REVIEW

File: stdlib/ballerina-builtin/src/main/java/org/ballerinalang/nativeimpl/io/channels/base/AbstractChannel.java [hunk 1]
```diff
@@ -174,4 +176,17 @@ public void close() {
...
     }
+
+    /**
+     * This will return {@link InputStream} from underlying {@link ByteChannel}.
+     *
+     * @return An {@link InputStream}
+     */
+    public InputStream getInputStream() {
+        if (!channel.isOpen()) {
+            String message = "Channel is already in a close state.";
```
Comment (pubudu91) [inline stdlib/ballerina-builtin/src/main/java/org/ballerinalang/nativeimpl/io/channels/base/AbstractChannel.java]: Shall we rephrase this to be simpler? i.e: "Channel is already closed."

File: tests/ballerina-test/src/test/java/org/ballerinalang/test/nativeimpl/functions/io/bytes/BytesInputOutputBufferTest.java [hunk 2]
```diff
@@ -206,4 +210,31 @@ public void writeBytesToFile() throws IOException {
...
+            final String result = buffer.lines().collect(Collectors.joining("\n"));
+            Assert.assertEquals(result, "123456", "Did not get expected string output.");
+        }
+        inputStream.close();
+    }
+
+    @Test(description = "Validate getting InputStream from closed channel",
+            expectedExceptions = BallerinaIOException.class,
+            expectedExceptionsMessageRegExp = "Channel is already closed.")
+    public void checkChanelCloseStatue() throws IOException, URISyntaxException {
```
Comment (chamil321) [inline tests/ballerina-test/src/test/java/org/ballerinalang/test/nativeimpl/functions/io/bytes/BytesInputOutputBufferTest.java]: Small typo? checkChannelCloseStatus() ?
